torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -365,8 +371,10 @@ std::shared_ptr<FutureMessage> TensorPipeAgent::send(
               }
 
               threadPool_.run(
```

Comment:
Unrelated, but is there a reason we defer this to the thread pool? Are we performing any blocking call?

Reply:
Yeah this seems like it shouldn't be deferred to the thread pool since we're just completing the future. cc @jiayisuse should we just do this in the main thread?